### PR TITLE
T34576 use p4a loading webview

### PR DESCRIPTION
**Pull request by dylanmccall**
https://phabricator.endlessm.com/T34576

-----

This requires that we merge endlessm/python-for-android#17, first. Note that there is a WIP commit (5ff4729f0ba80e49e875f42f798ed5358ead6d4a) which can be dropped once we have done that.

### CODE REVIEW

File: src/java/org/learningequality/FullScreen.java [hunk 1]
```diff
@@ -41,37 +44,55 @@ public static FullScreen getInstance() {
...
+    public void setAppKeyCookie(String url, String appKey) {
+        CookieManager.getInstance().setCookie(url, "app_key_cookie=" + appKey);
     }
 
     // Configure the WebView to allow fullscreen based on:
     // https://stackoverflow.com/questions/15768837/playing-html5-video-on-fullscreen-in-android-webview/56186877#56186877
     public void configure(final Runnable startWithNetwork, final Runnable startWithUSB, final Runnable loadingReady) {
-        mWebView.setWebContentsDebuggingEnabled(true);
-        mWebView.setWebViewClient(new WebViewClient() {
+        Log.i("Endless Key", "FullScreen configure");
```
Comment (manuq) [inline src/java/org/learningequality/FullScreen.java]: Nit: Maybe we can make "Endless Key" a constant:
```
private static final String TAG = "Endless Key";

...

Log.i(TAG, "FullScreen configure");

```

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 cython
 virtualenv
-git+https://github.com/endlessm/python-for-android@v2022.09.04-endless8#egg=python-for-android
+git+https://github.com/endlessm/python-for-android@51f2107a62248feb30f3d21338674fbfb6176184#egg=python-for-android
```
Comment (manuq) [inline requirements.txt]: 2nd commit is great for testing the p4a changes along with these ones in our CI build! I will approve without merging so you can update this line after tagging in the p4a repo.

File: src/kolibri_android/main_activity/activity.py [hunk 3]
```diff
@@ -35,8 +37,23 @@ def replace_url_in_webview(url):
...
+
+
+@Runnable
+def evaluate_javascript_in_loading_webview(js_code):
+    FullScreen.getInstance().mLoadingWebView.evaluateJavascript(js_code, None)
+
+
+@Runnable
+def evaluate_javascript_in_main_webview(js_code):
+    FullScreen.getInstance().mMainWebView.evaluateJavascript(js_code, None)
```
Comment (manuq) [inline src/kolibri_android/main_activity/activity.py]: There aren't any users of this one. Let's remove it for now.

File: src/kolibri_android/main_activity/activity.py [hunk 4]
```diff
@@ -35,8 +37,23 @@ def replace_url_in_webview(url):
...
 @Runnable
-def evaluate_javascript(js_code):
-    PythonActivity.mWebView.evaluateJavascript(js_code, None)
+def show_loading_page(url):
+    FullScreen.getInstance().showLoadingPage(url)
+
+
+@Runnable
+def evaluate_javascript_in_loading_webview(js_code):
+    FullScreen.getInstance().mLoadingWebView.evaluateJavascript(js_code, None)
```
Comment (manuq) [inline src/kolibri_android/main_activity/activity.py]: :+1:

File: src/kolibri_android/main_activity/activity.py [hunk 5]
```diff
@@ -145,7 +164,7 @@ def on_activity_save_instance_state(self, activity, out_state_bundle):
         logging.info(f"Saved Kolibri path: '{kolibri_path or ''}'")
 
     def _get_current_kolibri_path(self):
-        current_url = PythonActivity.mWebView.getUrl()
+        current_url = FullScreen.getInstance().getUrl()
```
Comment (manuq) [inline src/kolibri_android/main_activity/activity.py]: Nit: Maybe is a good time to rename "FullScreen" to a better name?

File: src/kolibri_android/main_activity/activity.py [hunk 6]
```diff
@@ -154,8 +173,7 @@ def _get_current_kolibri_path(self):
             return None
 
     def run(self):
-        self.replace_url("file:///android_asset/_load.html")
```
Comment (manuq) [inline src/kolibri_android/main_activity/activity.py:157]: :+1:

File: src/kolibri_android/main_activity/kolibri_bus.py [hunk 7]
```diff
@@ -65,12 +68,6 @@ def register_share_file_interface(share_file):
...
-        # Work around an issue where interface.get_initialize_url returns ""
-        # if next_url is empty. This is resolved in Kolibri v0.16.0-alpha8:
-        # <https://github.com/learningequality/kolibri/commit/0ed2ccdd5d613e96721f80bc03d8bc56ae7a0e7f>
-        # TODO: Remove this workaround when we update Kolibri.
-        next_url = self.application.get_saved_kolibri_path() or "/"
-        initialize_url = interface.get_initialize_url(next_url=next_url)
-        logging.info(f"Using initialize URL '{initialize_url}'")
-        start_url = urljoin(base_url, initialize_url)
+        next_url = self.application.get_saved_kolibri_path() or ""
+        start_url = urljoin(base_url, next_url)
```
Comment (manuq) [inline src/kolibri_android/main_activity/kolibri_bus.py:72]: :+1:

**Review by manuq — APPROVED** [review]

File: src/java/org/learningequality/FullScreen.java (hunk 1 above)
Comment (dylanmccall) [reply src/java/org/learningequality/FullScreen.java]: :+1:

File: src/kolibri_android/main_activity/activity.py (hunk 5 above)
Comment (dylanmccall) [reply src/kolibri_android/main_activity/activity.py]: Okay, after some deliberation I renamed it to `KolibriAndroidHelper`. Not a great name, but at least it isn't misleading :)

File: requirements.txt (hunk 2 above)
Comment (dylanmccall) [reply requirements.txt]: Okay, I changed this to refer to `v2022.09.04-endless9` in the newest push.

**Pull request merged**